src/QsCompiler/Core/Dependencies.fs

```diff
@@ -66,6 +71,21 @@ type BuiltIn = {
         | Value tId -> tId.Namespace.Value = GeneratedAttributes.Namespace && tId.Name.Value = GeneratedAttributes.LoadedViaTestNameInsteadOf
         | Null -> false
 
+    /// Returns the required runtime capability if the attribute is a valid instance of the RequiresCapability
+    /// attribute.
+    static member RequiredCapability attribute =
```

Comment:
There are already a couple of very similar methods at the beginning of the SymbolResolution module. It would make sense to follow the same pattern for everything; following that pattern, there would be a static member MarksRequiredCapability and then a separate TryGetRequiredCapability that takes a sequence of attributes. Probably it would make sense to split out both of these for all attributes that are recognized by the compiler into a separate module. That is an API change and we need to properly deprecate it, but seeing as the list of attributes that require similar methods grows, I think that would be a good change to make.

Reply:
I can see the advantage of changing this method to take in a sequence of attributes rather than a single attribute. I do not see the benefit of a separate `MarksRequiredCapability` method, though. I don't think it would be used anywhere.

Reply:
The benefit would be to stick with the same pattern for everything, and I think in terms of API predictability there is value to that. The idea is to have roughly the same methods for all compiler recognized attributes. I don't think that all Marks* methods could be eliminated (that would be the other option, but I think the C# generation might also be using one of these).

Reply:
I added `TryGetRequiredCapability`.

>The idea is to have roughly the same methods for all compiler recognized attributes. I don't think that all Marks* methods could be eliminated (that would be the other option, but I think the C# generation might also be using one of these).

The `Marks*` methods are written in a very repetitive way. Why not define one function that takes in a `BuiltIn` (or a `QsQualifiedName`) and an attribute and returns whether they match? That would work for all attributes, instead of needing one per attribute.

Reply:
I created a task to revise the API methods for compiler recognized attributes. Right now it is not very predictable.